### DISCUSSION TITLE
OpenId Connect Authorization policy one_factor/two_factor depending on internal/external network

### DISCUSSION THREAD

**Discussion by GuiPoM** [authelia/authelia#5637] [Ideas]
### Description
Support multiple authorization policies in OpenId Connect Authorization policy one_factor/two_factor depending on internal/external network 

### Use Case
I defined my access control so that I get a different rule depending if my user is connecting from internet or from my internal network. First cast, a two_factor is required, but otherwise a one_factor is secured enough. This is working nicely there.
I would like to define a similar configuration for oidc. If I connect from outside my internal network, authelia should also know it, and trigger a two_factor flow, otherwise a one_factor.

Does it also makes sense for other users ? I guess it would be a more homogeneous setting, as I currently need a two_factor within my own network if I first connect to a service defining oidc, when I bypass it when I connect first to a service only protected with an access rule.

**Comment by james-d-elliott** [comment]
The issue is that with the access policies they can be checked per-request whereas with OpenID Connect it can only be checked at the time of authorization/issuance which happens once in a tokens lifetime. This leads to the need of having OpenID Connect to handle this separately. It also means potentially more confusion regarding how it works.

I will leave this open and we'll figure out if/how we'll do this at some point though.

**Comment by djrm05** [comment]
is there any update on this? Thanks!

**Comment by james-d-elliott** [comment]
With lack of interest from users we will be making this into an idea discussion so users can clearly vote on it. It's important to note that there is no reasonable way to check the users IP address AFTER they have been given consent and the authorization has progressed. It may only be checked once, before the  authorization takes place. This means for the life of the token the user is theoretically able to skip this check when they move to another IP address.

If this feature is still desirable to users then we  recommend upvoting this discussion. Specifically this post here: https://github.com/authelia/authelia/discussions/5637?converting=1#discussion-5369606
